rpcs/blockService.go

```diff
@@ -382,7 +406,14 @@ func (bs *BlockService) rawBlockBytes(round basics.Round) ([]byte, error) {
 		}
 	default:
 	}
-	return RawBlockBytes(bs.ledger, round)
+	if bs.memoryUsed > bs.memoryCap {
+		return nil, errMemoryAtCapacity{used: bs.memoryUsed, capacity: bs.memoryCap}
+	}
+	data, err := RawBlockBytes(bs.ledger, round)
+	if err == nil {
+		bs.memoryUsed = bs.memoryUsed + uint64(len(data))
```

Comment:
I can see the clarity argument, since the subtraction is already done in ServerHTTP.

But, the main argument against your suggestion is the need for an extra locking needed. The lock is already acquired in `rawBlockBytes`, by moving this calculation to `ServerHTTP`, the lock needs to be acquired again after releasing it. 

Moreover, there are no side effects, since the change is on the `BlockService` object itself, not a global variable, and `rawBlockBytes` is a local function with only a single call site.

Reply:
I think this is the proper way of incrementing the counter. 
Moving this out is wrong. 

The decision to allocate the memory or return an error and the counter increment should be atomic.
If the increment is outside `rawBlockBytes` (not under the same lock session), and there are `n` parallel requests, then there can be `n` simultaneous check and allocations before the counter increments takes place.